src/components/DocExplorer.js

```diff
@@ -7,38 +7,92 @@
  */
 
 import React from 'react';
-
+import Marked from 'marked';
+import {
+  GraphQLUnionType,
+  GraphQLInterfaceType
+} from 'graphql/type';
 
 /**
  * DocExplorer
  *
+ * Shows documentations for GraphQL definitions from the schema.
  *
  * Props:
  *
+ *   - schema: A required GraphQLSchema instance that provides GraphQL document
+ *     definitions.
+ *
+ *   - typeName: An optional prop provided when a type is clicked from
+ *     QueryEditor hint object. The click triggers the document explorer to
+ *     show a definition of the type clicked.
  *
  */
 export class DocExplorer extends React.Component {
   constructor() {
     super();
 
-    this.EXPLORER_WIDTH = 400;
-    this.currentlyInspectedType = null;
+    this.EXPLORER_WIDTH = 450;
     this.state = {
       width: 'initial',
-      expanded: false
+      expanded: false,
+      inspectedType: null,
+      inspectedCall: null
     };
 
-    this.startPage = 'Welcome to GraphQL Documentation Explorer!';
+    this.startPage = '';
     this.content = '';
+
+    // Navigation stack to remember the definitions visited.
+    this.navStack = [];
+
+    // A static `Main Page` button that does not have to change.
+    this.backToMainButton = (
+      <button className="doc-back-to-main-button"
+        style={{ marginLeft: '6px' }}
+        onClick={this._onBackToMainBtnClick.bind(this)}
+      >
+        Main Page
+      </button>
+    );
+
+    this.marked = Marked;
+  }
+
+  _renderMarkdown(text) {
+    return <span
+      dangerouslySetInnerHTML={
+        {__html: this.marked(text)}
+      }
+    />;
+    return this.marked(text);
+  }
+
+  _getTypeLink(type) {
+    return type.ofType ?
+      <span>
+        [
+          <a className="doc-type">{type.ofType.name}</a>
```

Comment:
This assumes every wrapper type is a list. There's also the NonNull wrapper and the combinations of these two.
E.g. if your schema has

```js
fields: {
  foo: { type: new GraphQLNonNull(GraphQLString) },
  bar: { type: new GraphQLList(GraphQLString) },
  baz: { type: new GraphQLNonNull(new GraphQLList(GraphQLString)) },
}
```
this would render:
```
foo : [String]
bar : [String]
baz : [String]
```
The expected result would be:
```
foo : String!
bar : [String]
baz : [String]!
```